spec/inputs/google_pubsub_spec.rb

```diff
@@ -33,5 +34,9 @@
     expect {
       plugin = LogStash::Inputs::GooglePubSub.new(bad3)
     }.to raise_error(LogStash::ConfigurationError)
+    expect {
```

Comment:
I would make the statement of error that is tested more explicit. For example separating in its own `it` section like:

```ruby
  it "raise a configuration error if both 'json_key_file' and 'json_key_file_content' are present" do
    expect { 
      plugin = LogStash::Inputs::GooglePubSub.new(bad4)
      plugin.register()
    }.to raise_error(LogStash::ConfigurationError)
  end
```

and maybe rename `bad4` in `file_and_content_config_settings`